src/test/obj_ulog_size/obj_ulog_size.c

```diff
@@ -177,76 +203,100 @@ do_clt_overhead_test(PMEMobjpool *pop)
 	UT_ASSERTeq(ro_value, TX_INTENT_LOG_BUFFER_OVERHEAD);
 }
 
+/*
+ * do_tx_max_alloc_no_prealloc_snap -- fills the pool and tries to do
+ *		snapshot which is bigger than ulog size
+ */
 static void
 do_tx_max_alloc_no_prealloc_snap(PMEMobjpool *pop)
 {
-	UT_OUT("no_prealloc_snap");
-	PMEMoid oid[MAX_OBJECTS];
+	UT_OUT("do_tx_max_alloc_no_prealloc_snap");
+	PMEMoid *oids = NULL;
+	size_t noids = 0;
 
-	fill_pool(pop, oid);
+	oids = fill_pool(pop, &noids);
 
-	/* pool is full now, so let's try to snapshot first object */
-	size_t snap_size = pmemobj_alloc_usable_size(oid[0]);
+/*
+ * querying snapshot size
+ * oid[0] object is taken intentionally - size of the pool is 16 MB,
+ * allocation starts with objects with size of 1 MB, that's mean that
+ * oid[0] is bigger than default undo log, which size is 2 KB
+ */
+	size_t snap_size =
+		pmemobj_alloc_usable_size(oids[PREALLOCATED_OBJECT]);
 	TX_BEGIN(pop) {
 		/* it should abort - cannot allocate memory */
-		pmemobj_tx_add_range(oid[0], 0, snap_size);
+		pmemobj_tx_add_range(oids[PREALLOCATED_OBJECT], 0, snap_size);
 	} TX_ONABORT {
 		UT_OUT("!Cannot add snapshot");
 	} TX_ONCOMMIT {
 		UT_FATAL("Can add snapshot");
 	} TX_END
 
-	free_pool(oid);
+	free_pool(oids, noids);
 }
 
+/*
+ * do_tx_max_alloc_prealloc_snap -- fills the pool, apends preallocated
+ *		buffer and tries to do snapshot which is bigger than ulog size
+ */
 static void
 do_tx_max_alloc_prealloc_snap(PMEMobjpool *pop)
 {
-	UT_OUT("prealloc_snap");
-	PMEMoid oid[MAX_OBJECTS];
+	UT_OUT("do_tx_max_alloc_prealloc_snap");
+	PMEMoid *oids = NULL;
+	size_t noids = 0;
 
-	fill_pool(pop, oid);
+	oids = fill_pool(pop, &noids);
 
-	/* pool is full now, so let's try to snapshot first object */
-	size_t snap_size = pmemobj_alloc_usable_size(oid[0]);
-	void *addr = pmemobj_direct(oid[0]);
+	/* querying snapshot size */
+	size_t snap_size =
+		pmemobj_alloc_usable_size(oids[PREALLOCATED_OBJECT]);
+	void *addr = pmemobj_direct(oids[PREALLOCATED_OBJECT]);
 
 	TX_BEGIN(pop) {
 		pmemobj_tx_log_append_buffer(
 			TX_LOG_TYPE_SNAPSHOT, addr, snap_size);
-		pmemobj_tx_add_range(oid[0], 0, snap_size);
+		pmemobj_tx_add_range(oids[SNAPSHOT_OBJECT], 0, snap_size);
 	} TX_ONABORT {
 		UT_FATAL("!Cannot add snapshot");
 	} TX_ONCOMMIT {
 		UT_OUT("Can add snapshot");
 	} TX_END
 
-	free_pool(oid);
+	free_pool(oids, noids);
 }
 
+/*
+ * do_tx_max_alloc_prealloc_nested -- example of buffer appending
+ *		allocated by user in nested transaction
+ */
 static void
 do_tx_max_alloc_prealloc_nested(PMEMobjpool *pop)
 {
-	UT_OUT("prealloc_nested");
-	PMEMoid oid[MAX_OBJECTS];
+	UT_OUT("do_tx_max_alloc_prealloc_nested");
+	PMEMoid *oids = NULL;
+	size_t noids = 0;
 
-	fill_pool(pop, oid);
+	oids = fill_pool(pop, &noids);
 
-	size_t snap_size = pmemobj_alloc_usable_size(oid[1]);
-	void *addr = pmemobj_direct(oid[1]);
+	size_t snap_size =
```

Comment:
Change snap_size to buff_size, addr to buff_addr (those variables use in the pmemobj_alloc_usable_size function)
(Apply everywhere)